### PR TITLE
Refactor by pep8

### PR DESCRIPTION
**Pull request by zsef123**
Fixes https://github.com/RaRe-Technologies/gensim/issues/1521

First checkbox 
> PEP8 (without a hard 80-char constraint for long lines)

Most changes are whitespace adjustment and import order changes.

Then unused var and header are removed except in try catch, named _ var and side effects call. 
Excepted case add comments 'noqa'

And test_atmodel.py, test_ldamallet_wrapper.py, test_ldamodel codes don't use fname like under code
```
fname = testfile()
model = self.model
model.save(testfile(), sep_limit=0)
```
save or load argument changed from testfile() to fname

### CODE REVIEW

File: gensim/corpora/__init__.py [hunk 1]
```diff
@@ -3,15 +3,15 @@
...
-from .bleicorpus import BleiCorpus
-from .svmlightcorpus import SvmLightCorpus
-from .lowcorpus import LowCorpus
-from .dictionary import Dictionary
-from .hashdictionary import HashDictionary
-from .wikicorpus import WikiCorpus
-from .textcorpus import TextCorpus
-from .ucicorpus import UciCorpus
-from .malletcorpus import MalletCorpus
+from .mmcorpus import MmCorpus  # noqa:F401
```
Comment (piskvorky) [inline gensim/corpora/__init__.py]: -1: we don't want to litter the code with some tool-specific constructs.

Instead, open an issue with the tool that requires this, it's a bug on their side.

If you find such comments in other parts of gensim, please remove them.

File: gensim/examples/dmlcz/sources.py [hunk 2]
```diff
@@ -190,13 +187,13 @@ class DmlCzSource(DmlSource):
 
     See the ArticleSource class for general info on sources.
     """
+
     def idFromDir(self, path):
         assert len(path) > len(self.baseDir)
         dmlczId = open(os.path.join(path, 'dspace_id')).read().strip()
-        pathId = path[len(self.baseDir) + 1 : ]
+        pathId = path[len(self.baseDir) + 1:]
```
Comment (piskvorky) [inline gensim/examples/dmlcz/sources.py]: This is against PEP8 (see https://www.python.org/dev/peps/pep-0008/#other-recommendations ).

File: gensim/interfaces.py [hunk 3]
```diff
@@ -267,7 +263,7 @@ def __iter__(self):
                 # (unlike numpy). so, clip the end of the chunk explicitly to make
                 # scipy.sparse happy
                 chunk_end = min(self.index.shape[0], chunk_start + self.chunksize)
-                chunk = self.index[chunk_start : chunk_end]
+                chunk = self.index[chunk_start: chunk_end]
```
Comment (piskvorky) [inline gensim/interfaces.py]: PEP8: https://www.python.org/dev/peps/pep-0008/#other-recommendations (here and elsewhere).

File: gensim/corpora/__init__.py (hunk 1 above)
Comment (menshikh-iv) [reply gensim/corpora/__init__.py]: @piskvorky it's not a tool bug, it's really "unused import" by PEP

Comment (piskvorky) [reply gensim/corpora/__init__.py]: But it is used -- it makes the classes available directly from the package. Without it, you won't be able to `from corpora import MmCorpus`.

Or how can this be done in other way, that doesn't trigger F401?

Comment (menshikh-iv) [reply gensim/corpora/__init__.py]: > it makes the classes available directly from the package 
You are right, but now in the code, it's unused import.

I know another way - use this imports (like your example) in gensim internally, but I don't think that this is a good idea. For my opinion, `noqa` suppress is OK.

**Comment by zsef123** [comment]
> E722 | do not use bare except, specify exception instead

For example

```
try:
    "do something"
except:
    pass
```
E722 error occurs except line

In gensim code, some line add Exception class but the others not specify exception

So if not used exception class in exception scope, I just add Exception like this code

```
try:
    "do something"
except Exception:
    pass
```

**Comment by zsef123** [comment]
I miss check in corpora dir files.

I'm not touch about noqa, line slice whitespace yet.

and naming like 'l' changed to suitable name

File: gensim/matutils.py [hunk 4]
```diff
@@ -150,7 +150,7 @@ def zeros_aligned(shape, dtype, order='C', align=128):
     nbytes = np.prod(shape, dtype=np.int64) * np.dtype(dtype).itemsize
     buffer = np.zeros(nbytes + align, dtype=np.uint8)  # problematic on win64 ("maximum allowed dimension exceeded")
     start_index = -buffer.ctypes.data % align
-    return buffer[start_index: start_index + nbytes].view(dtype).reshape(shape, order=order)
+    return buffer[start_index:start_index + nbytes].view(dtype).reshape(shape, order=order)
```
Comment (piskvorky) [inline gensim/matutils.py]: Does not conform to PEP8 (need spaces around `:` here, see https://github.com/RaRe-Technologies/gensim/issues/1521#issuecomment-325270209).

Comment (zsef123) [reply gensim/matutils.py]: Sorry, Line slice white space was not my intention. I missed these line in this commit.

Anyway, I agree to input space around `:`.

but if input space like `return buffer[start_index : start_index + nbytes]`, E203 error occurs. [(pycodestyle issue)](https://github.com/PyCQA/pycodestyle/issues/373)

Is it okay upper line?

Comment (piskvorky) [reply gensim/matutils.py]: I don't think there's any error in the code, the version with spaces is correct. It appears to be a bug in the style-checking software.

**Comment by zsef123** [comment]
Issue https://github.com/RaRe-Technologies/gensim/issues/1551

> By adding # flake8: noqa to the file  [(docs:ignoring-entire-file)](http://flake8.pycqa.org/en/latest/user/violations.html#ignoring-entire-file)

I have adding above comment ignore file

It may be dangerous but I think `__init__.py` files short and simple enough so I can safely ignore flake8 errors 

Excepts [models/__init_.py](https://github.com/RaRe-Technologies/gensim/blob/develop/gensim/models/__init__.py), this file contains short class.
Still, I think it's okay to ignore the error because the file is still short, and support by PEP8.

File: gensim/corpora/__init__.py [hunk 5]
```diff
@@ -1,17 +1,17 @@
 """
 This package contains implementations of various streaming corpus I/O format.
 """
-
+# flake8: noqa
```
Comment (piskvorky) [inline gensim/corpora/__init__.py]: This seems to combine the worst of both worlds. Still extra comments, AND no style checking.

I think I prefer even the `noqa:F401` version.

**Comment by zsef123** [comment]
I canceled commit about ignore file.

> F403 : `from module import *` used; unable to detect undefined names

And just `from module import *` disassembled to each functions

**Comment by zsef123** [comment]
Sorry, I clicked delete accidently.

And I think I've finished applying PEP8.

1. white space adjustment
2. delete unused var( except naming `_` ), unused import modules
3. Add specific exceptions( `except Exception:` )
4. `noqa` comments about flake8 F401 error ( Issue https://github.com/RaRe-Technologies/gensim/issues/1551 )
5. disassembled `import *` about flake8 F403 error

**Comment by menshikh-iv** [comment]
Hi @zsef123, please remove file gensim/scripts/make_wiki_lemma.py from your PR ASAP (because you broke current symlink -> we can't build package)

**Comment by menshikh-iv** [comment]
@zsef123 I means to remove this file from your PR (not from the repo), you need to checkout this file to `develop` state.

**Comment by zsef123** [comment]
@menshikh-iv Sorry I check again.
But PEP8 E203 error occurs in travis
```
gensim/examples/dmlcz/sources.py:125:41: E203 whitespace before ':'
        intId = path[1 + path.rfind('#') :]

gensim/models/word2vec.py:1552:37: E203 whitespace before ':'
                        yield line[i : i + self.max_sentence_length]
```
Issue https://github.com/RaRe-Technologies/gensim/issues/1521
>Second, line changed to
return buffer[start_index:start_index + nbytes].view(dtype).reshape(shape, order=order)
these line not occurs error in pep8 library and fine too

And

>piskvorky commented 8 days ago • edited
The second option does not conform to PEP8 (see the "No:" section in https://www.python.org/dev/peps/pep-0008/#pet-peeves).Basically, the : operator has a lower priority than e.g. +, and so shouldn't have less whitespace because that implies more priority in Python, by convention.

So I input space around `:`. Is it problem?

**Comment by menshikh-iv** [comment]
@zsef123 space **before** `:` is problem here.

**Comment by zsef123** [comment]
In matutils.py, I already remove space before `:` ( in first commits )
And I requested like under  code
```
-    return buffer[start_index: start_index + nbytes].view(dtype).reshape(shape, order=order)
+    return buffer[start_index:start_index + nbytes].view(dtype).reshape(shape, order=order)
```

>zsef123 7 days ago • edited Contributor
Sorry, Line slice white space was not my intention. I missed these line in this commit.
Anyway, I agree to input space around :.
but if input space like return buffer[start_index : start_index + nbytes], E203 error occurs. (pycodestyle issue)
Is the above line okay if an error occurs?

>piskvorky 7 days ago • edited Owner
I don't think there's any error in the code, the version with spaces is correct. The version without spaces is incorrect. It appears to be a bug in the style-checking software. CC @menshikh-iv

And Issue https://github.com/RaRe-Technologies/gensim/issues/1521

>The second option does not conform to PEP8 (see the "No:" section in https://www.python.org/dev/peps/pep-0008/#pet-peeves).Basically, the : operator has a lower priority than e.g. +, and so shouldn't have less whitespace because that implies more priority in Python, by convention.

So I changed to input space around `:`

**Comment by zsef123** [comment]
In matutils.py, I already remove space before `:` ( in first commits )
And I requested like under  code
```
-    return buffer[start_index: start_index + nbytes].view(dtype).reshape(shape, order=order)
+    return buffer[start_index:start_index + nbytes].view(dtype).reshape(shape, order=order)
```

>zsef123 7 days ago • edited Contributor
Sorry, Line slice white space was not my intention. I missed these line in this commit.
Anyway, I agree to input space around :.
but if input space like return buffer[start_index : start_index + nbytes], E203 error occurs. (pycodestyle issue)
Is the above line okay if an error occurs?

>piskvorky 7 days ago • edited Owner
I don't think there's any error in the code, the version with spaces is correct. The version without spaces is incorrect. It appears to be a bug in the style-checking software. CC @menshikh-iv

So I changed to input space around `:`

**Comment by zsef123** [comment]
@menshikh-iv Should I remove space before `:`? like `line[i: i + self.max_sentence_length]` 
This line passed Travis

**Comment by menshikh-iv** [comment]
@zsef123 yep

**Comment by menshikh-iv** [comment]
Very large and useful work, thank you very much @zsef123 :fire:

**Pull request merged**